packages/framework/block-std/src/gfx/selection.ts

```diff
@@ -0,0 +1,390 @@
+import {
+  assertType,
+  DisposableGroup,
+  getUnitedBound,
+  groupBy,
+  type IPoint,
+  Slot,
+} from '@blocksuite/global/utils';
+
+import type { CursorSelection, SurfaceSelection } from '../selection/index.js';
+import type { GfxModel } from './gfx-block-model.js';
+
+import {
+  type GfxController,
+  GfxExtension,
+  GfxExtensionIdentifier,
+} from './controller.js';
+import { GfxGroupLikeElementModel } from './surface/element-model.js';
+
+export interface SurfaceSelectionState {
+  /**
+   * The selected elements. Could be blocks or canvas elements
+   */
+  elements: string[];
+
+  /**
+   * Indicate whether the selected element is in editing mode
+   */
+  editing?: boolean;
+
+  /**
+   *  Cannot be operated, only box is displayed
+   */
+  inoperable?: boolean;
+}
+
+/**
+ * GfxSelectionManager is just a wrapper of std selection providing
+ * convenient method and states in gfx
+ */
+export class GfxSelectionManager extends GfxExtension {
+  static override key = 'gfxSelection';
+
+  private _activeGroup: GfxGroupLikeElementModel | null = null;
+
+  private _cursorSelection: CursorSelection | null = null;
+
+  private _lastSurfaceSelections: SurfaceSelection[] = [];
+
+  private _remoteCursorSelectionMap = new Map<number, CursorSelection>();
+
+  private _remoteSelectedSet = new Set<string>();
+
+  private _remoteSurfaceSelectionsMap = new Map<number, SurfaceSelection[]>();
+
+  private _selectedSet = new Set<string>();
+
+  private _surfaceSelections: SurfaceSelection[] = [];
+
+  disposable: DisposableGroup = new DisposableGroup();
+
+  readonly slots = {
```

Comment:
It makes more sense to use signals instead of slots here because you're watching reactive data instead of handling events.

Reply:
I just copy from the old file, the refactor of part haven't start yet.